### PR TITLE
Initial transparent long stack trace support

### PR DESCRIPTION
**Pull request by CaptEmulation**
unfortunately, some necessary unit tests which should be passing are failing to satisfy `2.2.7.2 f either onFulfilled or onRejected throws an exception e, promise2 must be rejected with e as the reason.` from https://promisesaplus.com/

### CODE REVIEW

**Comment by neonphog** [comment]
created https://github.com/CaptEmulation/ppp/pull/1

File: lib/ppp.js [hunk 1]
```diff
@@ -160,6 +167,19 @@
...
+      try {
+        throw new Error();
+      } catch (e) {
+        // NOTE: don't try to use `Error.captureStackTrace` or transfer the
+        // accessor around; that causes memory leaks as per GH-111. Just
+        // reify the stack trace as a string ASAP.
+        //
+        // At the same time, cut off the first line; it's always just
+        // "[object Promise]\n", as per the `toString`.
+        this.stack = e.stack.substring(e.stack.indexOf("\n") + 1);
```
Comment (neonphog) [inline lib/ppp.js]: We may need to not do this... our version of opera doesn't include the error in the first line, so we might be deleting a line of stacktrace...

Comment (CaptEmulation) [inline lib/ppp.js]: I'll check

**Comment by CaptEmulation** [comment]
- Unit tests pass
 - Removed the stripping of the first line of exception due to Opera
 - Added internal stack stripping support for Chrome

Does not seem to work for minified code.  I don't see that as a problem.

**Comment by neonphog** [comment]
> Does not seem to work for minified code. I don't see that as a problem.

I'm not sure why, but after I tweaked a couple unrelated things, the minified stack traces started working. Maybe you forgot to `grunt uglify` after doing a fix?

Thanks for this, I think this is going to be super helpful!!

**Comment by CaptEmulation** [comment]
> Maybe you forgot to grunt uglify after doing a fix?

probably

**Pull request merged**